realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -112,6 +114,7 @@ public boolean handleMessage(Message message) {
             QueryUpdateTask.Result result;
             switch (message.what) {
 
+                case LOCAL_COMMIT:
```

Comment:
```
    private void realmChanged() {
        deleteWeakReferences();
        if (threadContainsAsyncQueries()) {
            updateAsyncQueries();

        } else {
            RealmLog.d("REALM_CHANGED realm:" + HandlerController.this + " no async queries, advance_read");
            realm.sharedGroupManager.advanceRead();
            notifyAllListeners();
        }
    }
```

`if (threadContainsAsyncQueries()) {` is not covered ... @#$!@#$@!$# This hurts my brain too much !!!